### PR TITLE
Create example notebook that lets users input plasma properties and get plasma parameters

### LINKED ISSUE #939: Create example notebook that lets users input plasma properties and get plasma parameters
It's pretty common for plasma physicists to have a bunch of plasma properties (e.g., magnetic field strength, ion/electron number densities, etc.) and want to calculate a bunch of plasma parameters (e.g., Debye length, electron & ion inertial lengths, electron & ion gyroradii and gyrofrequencies, etc.).  It would be really helpful to add a notebook in our [examples](https://docs.plasmapy.org/en/latest/examples.html) that allows users to input different plasma properties and then get a whole bunch of these plasma parameters printed out.  This notebook would use functionality in [`plasmapy.formulary`](https://docs.plasmapy.org/en/latest/formulary/index.html).

It would probably help to provide the inverses of different frequencies to get the associated time scales (i.e. the inverse of the gyrofrequency will give the gyroperiod).  The different length scales and time scales should be placed near each other.  It'd also be helpful to sort the different length and time scales from shortest to longest.  

This notebook would complement the notebook on [calculating ITER parameters](https://docs.plasmapy.org/en/latest/notebooks/physics.html) (which could probably use some updates too, since its obsolete reference to the `physics` subpackage should be changed to a reference to the `formulary` subpackage).  

Thanks to @svincena for suggesting this idea!

### PR DESCRIPTION
**Pull request by RAJAGOPALAN-GANGADHARAN**
Closes #939

An interactive notebook that lets users input plasma properties and get various plasma parameters.

- [x] I have added a changelog entry for this pull request.

### CODE REVIEW

**Comment by RAJAGOPALAN-GANGADHARAN** [comment]
This is the current state of notebook.
calc

**Comment by RAJAGOPALAN-GANGADHARAN** [comment]
Requesting review from @namurphy

**Comment by namurphy** [comment]
@RAJAGOPALAN-GANGADHARAN — Thank you for working on this!  The widget is looking pretty nice so far.  I haven't used widgets much before, so it's really interesting to see how you're setting this up.

The main two things I'm wondering about are:
 - How can we get this to render in our online documentation as an [example notebook](https://docs.plasmapy.org/en/latest/examples.html)?  
 - How should we handle the behind-the-scenes code to set up the widgets?

There are a couple of approaches that I'm thinking about, though I'm not sure what the best approach would be.  One possibility would be to put the widget setup code into a separate file and then import the app.  If we aren't able to get the app to render correctly in our online documentation, an alternative would be to set up a binder environment based in a separate GitHub repository, and then link to that environment from a few places in the documentation.   

Right now there an error in the `CI / Documentation (pull_request)` test that says that that each notebook needs at least one header, so probably the next step would be to add a Markdown cell with `# Calculate plasma parameters` at the top.  Then hopefully we'd be able to see how the notebook would render in the [examples gallery for this pull request](https://plasmapy--1229.org.readthedocs.build/en/1229/examples.html).  

Thank you again!

**Comment by RAJAGOPALAN-GANGADHARAN** [comment]
@namurphy I have tried using voila as @StanczakDominik suggested during the meeting. It works absolutely fine. Maybe we can package into an app (might have an electron app as well) and provide installations for the same. For now I created a script file with the command but couldn't figure out the best place to put it. Please check it out!

Thank you guys

**Review by namurphy — COMMENTED** [review]
My apologies for the delay!  Part of the reason has been that I've been uncertain about how best to include this in PlasmaPy or what API it should have.  I've been slowly coming around to the idea that, after a user does `pip install plasmapy`, that they would be able to type something like `plasma-calculator` in the command line and have this app launch in their web browser.  This is along the lines of `plasma_calculator.sh`.  I don't know [how to package a shell script with a Python package](https://www.google.com/search?channel=fs&client=ubuntu&q=how+to+package+a+shell+script+with+a+Python+package), but a quick web search suggests that there are ways to do this.  

That was my most general thought, and I had a few more specific thoughts in the PR.  If you think it'd be helpful to meet, feel free to use this [scheduling link](https://calendly.com/namurphy) which automates setting up a Zoom link and calendar event.  

Sorry again about the delay, and thank you once more for working on this!

**Comment by RAJAGOPALAN-GANGADHARAN** [comment]
Thank you @namurphy I was creating the script based on this idea only, now it makes more sense to wrap it with the module. Will address the comments in the weekend. Thank you again :)

**Comment by RAJAGOPALAN-GANGADHARAN** [comment]
@namurphy So I incorporated the calculator as a tool that can be installed along side module using pip. Let me know if this model works fine

**Comment by RAJAGOPALAN-GANGADHARAN** [comment]
pre-commit.ci autofix

**Review by namurphy — COMMENTED** [review]
I'm still liking how it's turning out!  The docstrings are looking pretty good.  I have a few minor suggestions but nothing major.

I think the documentation build is failing since there needs to be a stub file created in the documentation for the module you created.  We just added some information about that in the last bullet point in the [last section of the documentation guide](https://docs.plasmapy.org/en/latest/development/doc_guide.html#narrative-documentation-guidelines).  

Thanks again!

**Comment by RAJAGOPALAN-GANGADHARAN** [comment]
pre-commit.ci autofix

**Comment by namurphy** [comment]
@RAJAGOPALAN-GANGADHARAN — This is looking great!  I'm finding the code to be quite readable which will be really helpful in the future.  

I have one more thought.  Right now `_GenericWidget` requires subclasses to implement a method called `create_widget`.  What would you think of using an [abstract base class](https://docs.python.org/3/library/abc.html) to codify this requirement?  It'd be something like

```Python
import abc

class _GenericWidget(abc.ABC):
    @abc.abstractmethod
    def create_widget(self):
        pass
```

**Comment by RAJAGOPALAN-GANGADHARAN** [comment]
@namurphy Thank you, I could use abstract base classes that does seem useful. On the docstring part - I will try my best to scout for changes.

**Comment by RAJAGOPALAN-GANGADHARAN** [comment]
Hello @namurphy added the abstract method enforcement and as a first step in the readme cleanup added periods to description sentences for consistencies. Let me know if something else needs change! Thanks :)

**Comment by RAJAGOPALAN-GANGADHARAN** [comment]
pre-commit.ci autofix

File: plasmapy/utils/calculator/widget_helpers.py [hunk 1]
```diff
@@ -0,0 +1,696 @@
...
+    @abc.abstractmethod
+    def create_widget(self):
+        """
+        Virtual method to create widget
+
+        Property
+        --------
+        `abstractmethod`
+            Protected by abc.abstractmethod wrapper to ensure the method is implemented.
+        """
```
Comment (rocco8773) [inline plasmapy/utils/calculator/widget_helpers.py:152]:
```suggestion
        """Virtual method to create widget"""
```

File: plasmapy/utils/calculator/__init__.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
+"""
+Script and utilities to launch the plasma calculator
+"""
+__all__ = [
+    "main",
+]
```
Comment (rocco8773) [inline plasmapy/utils/calculator/__init__.py:6]:
```suggestion
__all__ = ["main"]

```

File: plasmapy/utils/calculator/main_interface.py [hunk 3]
```diff
@@ -0,0 +1,174 @@
+"""
+Collection of private functions to load properties and construct widgets
+"""
+__all__ = [
+    "grid_data",
+    "LIGHT_BLUE",
+    "LIGHT_GRAY",
+    "_create_interactive_layout",
+    "_create_output_layout",
+]
```
Comment (rocco8773) [inline plasmapy/utils/calculator/main_interface.py:10]:
```suggestion
__all__ = []
```

File: plasmapy/utils/calculator/widget_helpers.py [hunk 4]
```diff
@@ -0,0 +1,696 @@
...
+    "_ParticleBox",
+    "_CheckBox",
+    "_IonBox",
+    "_FunctionInfo",
+    "_create_label",
+    "_handle_button_click",
+    "_handle_clear_click",
+    "_colored_text",
+    "_create_widget",
+]
```
Comment (rocco8773) [inline plasmapy/utils/calculator/widget_helpers.py:24]:
```suggestion
__all__ = []
```

**Comment by RAJAGOPALAN-GANGADHARAN** [comment]
@namurphy Addressed the review comments from yesterday's meetup. Thanks!

**Review by namurphy — APPROVED** [review]

**Comment by namurphy** [comment]
Just merged it.  Thank you again for doing this!

**Pull request merged**